File: workspace/judoku/src/main/java/judoku/checkers/CheckCellsForNoMorePotentialValuesAvailable.java
package judoku.checkers;

import judoku.data.Block;
import judoku.data.Cell;
import judoku.data.Grid;

public class CheckCellsForNoMorePotentialValuesAvailable implements IConsistencyCheck {
	@Override
	// check if there is a cell which is not fix but has no more potential values
	public boolean checkConsistency(Grid grid) {
		for (Block b: grid.getBlocks()) {
			for (Cell c: b.getCells()) {
				if (!(c.isFix()) && c.getPotentialValues().isEmpty()) {
					throw new RuntimeException("Cell c " + c.id + " is not fixed but has no more potential values");
				}
			}
		}
		return true;
	}
}

File: workspace/scudoku/src/main/scala/scudoku/checkers/CheckCellsForNoMorePotentialValuesAvailable.scala
package scudoku.checkers

/**
 * Java2Scala port from judoku.checkers.CheckCellsForNoMorePotentialValuesAvailable
 */

import scudoku.data._

class CheckCellsForNoMorePotentialValuesAvailable extends TConsistencyCheck {
  // check if there is a cell which is not fix but has no more potential values
  override def checkConsistency(grid: Grid): Boolean = {
    ! grid.blocks.
      map((block: Block) => block.getCells). 
      exists(cells => 
        cells.exists(cell => (!cell.isFix && cell.potentialValues.isEmpty)))
  }
}